__tests__/core/type-system.test.ts

```diff
@@ -1,5 +1,3 @@
-import { Exact } from "ts-essentials"
```

Comment:
I didn't realized it, but `Exact` from `ts-essentials` isn't really "exact"! It's just "if A and B are record types, the shapes of the two are the same", which is definitely not what we wanted. My bad!

(TS is hard)